### PR TITLE
add: chain_id param in etherscan query req

### PR DESCRIPTION
**Pull request by yash-atreya**
Required for foundry-rs/foundry#7485

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
hmm, do all the other endpoints now also support the chain id or just verification?

I think this only covers verification:

https://docs.etherscan.io/contract-verification/multichain-verification

so I think we should limit this pr to verify

**Comment by yash-atreya** [comment]
> hmm, do all the other endpoints now also support the chain id or just verification?
> 
> I think this only covers verification:
> 
> https://docs.etherscan.io/contract-verification/multichain-verification
> 
> so I think we should limit this pr to verify

I could just append `chain_id` to `other` in verify related methods in https://github.com/foundry-rs/block-explorers/blob/main/src/verify.rs

**Comment by mattsse** [comment]
we can add this as optional arg to the verify args

```rust 
#[serde(rename = "chainId", skip_serializing_if = "Option::is_none")]
    pub chain_id: Option<bool>,
```

File: src/lib.rs [hunk 1]
```diff
@@ -257,12 +259,14 @@ impl Client {
         &self,
         module: &'static str,
         action: &'static str,
+        chain_id: Option<u64>,
```
Comment (mattsse) [inline src/lib.rs:262]: this argument is redundant because we already have it on self.

File: src/lib.rs [hunk 2]
```diff
@@ -343,6 +350,12 @@ impl ClientBuilder {
         self
     }
 
+    /// Configures the chain id for etherscan verification
+    pub fn with_chain_id(mut self, chain_id: u64) -> Self {
```
Comment (mattsse) [inline src/lib.rs:354]: This should accept `Chain`
we also need a getter for this

File: src/lib.rs [hunk 3]
```diff
@@ -343,6 +350,12 @@ impl ClientBuilder {
         self
     }
 
+    /// Configures the chain id for etherscan verification
```
Comment (mattsse) [inline src/lib.rs:353]: let's link to etherscan multichain verification docs

File: src/lib.rs [hunk 4]
```diff
@@ -302,7 +308,8 @@ impl ClientBuilder {
             .etherscan_urls()
             .map(urls)
             .ok_or_else(|| EtherscanError::ChainNotSupported(chain))?;
-        self.with_api_url(etherscan_api_url?)?.with_url(etherscan_url?)
+
+        self.with_chain_id(chain.id()).with_api_url(etherscan_api_url?)?.with_url(etherscan_url?)
```
Comment (mattsse) [inline src/lib.rs:312]: we need to add a note to function docs that this also sets the chain id for multichain verification

**Review by mattsse — CHANGES_REQUESTED** [review]
great!
a few more nits

**Review by mattsse — APPROVED** [review]
lgtm, I believe this should do it

**Pull request merged**